internal/pkg/deploy/cloudformation/stack/lb_web_svc.go

```diff
@@ -260,6 +261,22 @@ func (s *LoadBalancedWebService) Template() (string, error) {
 	return string(overriddenTpl), nil
 }
 
+func (s *LoadBalancedWebService) getPrimaryContainerPortMappings() []*template.PortMapping {
```

Comment:
Is it possible that we let manifest do all of this:
```
func (s *LoadBalancedWebServiceManifest) portMappings() []template.PortMapping {
  // Read `image.port`
  // Read `http.target_port`
  // Read `sidecars.port`
  // Produce a list of port mappings where the port number is unique.
}
```

Reply:
Hmm, that sounds interesting idea to me 😍

Reply:
Yay!  Wanxian's feedback looks coincidentally like my comment above.

My slight tweak is that I don't think the `manifest` pkg should depend on `template` types.
The `template` types like `template.PortMapping` in my mind are tightly coupled with CFN/AWS API.
Whereas the `manifest` methods should return I think its own types that can then be converted to other data types, like CFN or Terraform or ...